spec/formatting.md

```diff
@@ -19,6 +19,111 @@ the local variable takes precedence.
 It is an error for a local variable definition to
 refer to a local variable that's defined after it in the message.
 
+## Pattern Selection
+
+When formatting a message with one or more selectors,
+the Pattern of one of Variants must be selected for formatting.
+
+An implementation may use any pattern selection method,
+as long as its observable behaviour matches the results of the method defined here.
+
+### Resolve Selectors
+
+First, resolve the values of each Selector Expression:
+
+1. Let _res_ be a new empty list of resolved values that support selection.
+2. For each Expression _exp_ of the message's Selector Expressions,
+   1. Let _rv_ be the resolved value of _exp_.
+   2. If selection is supported for _rv_:
+      1. Append _rv_ as the last element of the list _res_.
+   3. Else:
+      1. Let _nomatch_ be a resolved value for which selection always fails.
+      2. Append _nomatch_ as the last element of the list _res_.
+      3. Emit a Selection Error.
+
+The shape of the resolved values must be determined by each implementation,
+along with the manner of determining their support for selection.
+
+### Select Pattern
+
+Using _res_,
+the Variants are iterated in source order and the following test is performed
+to find one with all of its keys matching the Selector Expressions:
+
+1. For each index _i_ in _res_:
+   1. Let _key_ be the VariantKey at position _i_.
+   2. If _key_ is not equal to the catch-all key `'*'`:
+      1. Let _sel_ be the entry in _res_ at position _i_.
+      2. Let _pass_ be the boolean result of testing _key_ against _sel_.
+      3. If _pass_ is False,
+         1. Return False to indicate failure.
+2. Return True to indicate success.
+
+The manner of testing _key_ against _sel_ must be defined by each implementation.
```

Comment:
This is the place in the text that I was referring to in the earlier comment. It seems to suggest that the matching / testing of the runtime value element `_sel_` to the element `_key_` of the VariantKey, should be defined by the implementation, which sounds like implementation of MF2.  If not, then the wording is confusing.

And regardless, we still want to be more specific about what part of MF2 is responsible for the matching logic of `_sel_` to `_key_` since we know from plurals that we can't assume the trivial case (value equality comparison) to be sufficient.